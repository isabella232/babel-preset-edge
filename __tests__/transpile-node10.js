import buildPreset from "../src"
import { check, fixtures, titles } from "./core"

describe("Transpile: NodeJS v8", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    target: "node",
    transpile: { node: 10 }
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
