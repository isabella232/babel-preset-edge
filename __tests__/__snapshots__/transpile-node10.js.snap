// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpile: NodeJS v8 comments 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _camelCase2 = _interopRequireDefault(require(\\"lodash/camelCase\\"));

function translate(value) {
  return (0, _camelCase2.default)(value);
}"
`;

exports[`Transpile: NodeJS v8 deadcode 1`] = `
"\\"use strict\\";

const STATIC = false;
before();
after();"
`;

exports[`Transpile: NodeJS v8 emotion 1`] = `
"\\"use strict\\";

var _emotion = require(\\"emotion\\");

const className =
/*#__PURE__*/
(0, _emotion.css)({
  name: \\"1vdp66k-className\\",
  styles: \\"color:hotpink;label:className;\\"
});

function SomeComponent() {
  return React.createElement(\\"div\\", {
    className: className
  }, \\"Some hotpink text.\\");
}"
`;

exports[`Transpile: NodeJS v8 es2015: classes 1`] = `
"\\"use strict\\";

exports.__esModule = true;
exports.Component = void 0;

class Component extends Other {
  constructor() {
    super();
    this.flag = true;
  }

  main() {
    alert(this.foo);
  }

}

exports.Component = Component;"
`;

exports[`Transpile: NodeJS v8 es2015: for_off 1`] = `
"\\"use strict\\";

function test() {
  for (const taskName of tasks) {
    try {
      executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Transpile: NodeJS v8 es2015: generator 1`] = `
"\\"use strict\\";

function* idMaker() {
  var index = 0;

  while (true) yield index++;
}"
`;

exports[`Transpile: NodeJS v8 es2015: set 1`] = `
"\\"use strict\\";

new Set([1, 2, 3]);"
`;

exports[`Transpile: NodeJS v8 es2016: exponentiation 1`] = `
"\\"use strict\\";

var result = 2 ** 3;"
`;

exports[`Transpile: NodeJS v8 esnext: async 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.promise\\");

async function wait() {
  await loadData();
}"
`;

exports[`Transpile: NodeJS v8 esnext: async_loop 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.promise\\");

async function executeTasks(tasks) {
  for (const taskName of tasks) {
    try {
      await executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Transpile: NodeJS v8 esnext: catch_bind 1`] = `
"\\"use strict\\";

try {
  throw 0;
} catch (_unused) {
  doSomethingWhichDoesntCareAboutTheValueThrown();
} finally {
  doSomeCleanup();
}"
`;

exports[`Transpile: NodeJS v8 esnext: class properties 1`] = `
"\\"use strict\\";

exports.__esModule = true;
exports.Component = void 0;

class Component {
  constructor() {
    this.onClick = () => {};
  }

}

exports.Component = Component;"
`;

exports[`Transpile: NodeJS v8 esnext: decorators 1`] = `
"\\"use strict\\";

var _class;

let MyClass = annotation(_class = class MyClass {}) || _class;

function annotation(target) {
  target.annotated = true;
}"
`;

exports[`Transpile: NodeJS v8 esnext: dynamic import 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.promise\\");

import(
/*webpackChunkName:'lodash-wBYk8'*/
\\"lodash\\").then(lodash => {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`Transpile: NodeJS v8 esnext: object spread 1`] = `
"\\"use strict\\";

let original = {
  bar: 10
},
    variant = Object.assign({
  foo: 1
}, original);"
`;

exports[`Transpile: NodeJS v8 loadable 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.promise\\");

exports.__esModule = true;
exports.Contact = exports.About = exports.Home = void 0;

var _loadableComponents = _interopRequireDefault(require(\\"loadable-components\\"));

const Home = (0, _loadableComponents.default)({
  chunkName() {
    return \\"Home-JfPYC\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Home-JfPYC\\" */
  \\"./views/Home\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return require('path').resolve(__dirname, \\"./views/Home\\");
  }

});
exports.Home = Home;
const About = (0, _loadableComponents.default)({
  chunkName() {
    return \\"About-8yDxR\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"About-8yDxR\\" */
  \\"./views/About\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return require('path').resolve(__dirname, \\"./views/About\\");
  }

});
exports.About = About;
const Contact = (0, _loadableComponents.default)({
  chunkName() {
    return \\"Contact-AIf6m\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Contact-AIf6m\\" */
  \\"./views/Contact\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return require('path').resolve(__dirname, \\"./views/Contact\\");
  }

});
exports.Contact = Contact;"
`;

exports[`Transpile: NodeJS v8 loadable magic-comments 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.promise\\");

exports.__esModule = true;
exports.Contact = exports.About = exports.Home = void 0;

var _loadableComponents = _interopRequireDefault(require(\\"loadable-components\\"));

const Home = (0, _loadableComponents.default)({
  chunkName() {
    return \\"Home-JfPYC\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Home-JfPYC\\" */
  \\"./views/Home\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return require('path').resolve(__dirname, \\"./views/Home\\");
  }

});
exports.Home = Home;
const About = (0, _loadableComponents.default)({
  chunkName() {
    return \\"About\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"About\\" */
  \\"./views/About\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return require('path').resolve(__dirname, \\"./views/About\\");
  }

});
exports.About = About;
const Contact = (0, _loadableComponents.default)({
  chunkName() {
    return \\"Contact-AIf6m\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackMode: \\"lazy\\", webpackFoo: \\"bar\\", webpackChunkName: \\"Contact-AIf6m\\" */
  \\"./views/Contact\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return require('path').resolve(__dirname, \\"./views/Contact\\");
  }

});
exports.Contact = Contact;"
`;

exports[`Transpile: NodeJS v8 loadable wildcard 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.promise\\");

exports.__esModule = true;
exports.reactIntl = exports.leanIntl = exports.views = void 0;

var _loadableComponents = _interopRequireDefault(require(\\"loadable-components\\"));

const page = \\"About\\",
      lang = \\"de_DE\\",
      views = (0, _loadableComponents.default)({
  chunkName() {
    return \`views-\${page}\`.replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"views-[request]\\" */
  \`./views/\${page}\`),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\`./views/\${page}\`);
    }

    return require('path').resolve(__dirname, \`./views/\${page}\`);
  }

});
exports.views = views;
const leanIntl = (0, _loadableComponents.default)({
  chunkName() {
    return \`lean-intl-\${lang}-data\`.replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"lean-intl-[request]-data\\" */
  \`./lean-intl/\${lang}/data\`),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\`./lean-intl/\${lang}/data\`);
    }

    return require('path').resolve(__dirname, \`./lean-intl/\${lang}/data\`);
  }

});
exports.leanIntl = leanIntl;
const reactIntl = (0, _loadableComponents.default)({
  chunkName() {
    return \`react-intl-\${lang}-data\`.replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"react-intl-[request]-data\\" */
  \`./react-intl/\${lang}/data\`),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\`./react-intl/\${lang}/data\`);
    }

    return require('path').resolve(__dirname, \`./react-intl/\${lang}/data\`);
  }

});
exports.reactIntl = reactIntl;"
`;

exports[`Transpile: NodeJS v8 lodash: async 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _parallel2 = _interopRequireDefault(require(\\"async/parallel\\"));

(0, _parallel2.default)([]);"
`;

exports[`Transpile: NodeJS v8 lodash: lodash 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _camelCase2 = _interopRequireDefault(require(\\"lodash/camelCase\\"));

(0, _camelCase2.default)(\\"hello world\\");"
`;

exports[`Transpile: NodeJS v8 lodash: recompose 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _compose2 = _interopRequireDefault(require(\\"recompose/compose\\"));

(0, _compose2.default)([]);"
`;

exports[`Transpile: NodeJS v8 macro: gql 1`] = `
"\\"use strict\\";

const query = {
  kind: \\"Document\\",
  definitions: [{
    kind: \\"OperationDefinition\\",
    operation: \\"query\\",
    name: {
      kind: \\"Name\\",
      value: \\"User\\"
    },
    variableDefinitions: [],
    directives: [],
    selectionSet: {
      kind: \\"SelectionSet\\",
      selections: [{
        kind: \\"Field\\",
        name: {
          kind: \\"Name\\",
          value: \\"user\\"
        },
        arguments: [{
          kind: \\"Argument\\",
          name: {
            kind: \\"Name\\",
            value: \\"id\\"
          },
          value: {
            kind: \\"IntValue\\",
            value: \\"5\\"
          }
        }],
        directives: [],
        selectionSet: {
          kind: \\"SelectionSet\\",
          selections: [{
            kind: \\"Field\\",
            name: {
              kind: \\"Name\\",
              value: \\"lastName\\"
            },
            arguments: [],
            directives: []
          }, {
            kind: \\"FragmentSpread\\",
            name: {
              kind: \\"Name\\",
              value: \\"UserEntry1\\"
            },
            directives: []
          }]
        }
      }]
    }
  }],
  loc: {
    start: 0,
    end: 79,
    source: {
      body: \\"\\\\n  query User {\\\\n    user(id: 5) {\\\\n      lastName\\\\n      ...UserEntry1\\\\n    }\\\\n  }\\\\n\\",
      name: \\"GraphQL request\\",
      locationOffset: {
        line: 1,
        column: 1
      }
    }
  }
};"
`;

exports[`Transpile: NodeJS v8 macro: preval 1`] = `
"\\"use strict\\";

const one = 1;"
`;

exports[`Transpile: NodeJS v8 promise 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.promise\\");

exports.__esModule = true;
exports.default = _default;

function _default() {
  return new Promise(resolve => {
    return resolve(42);
  });
}"
`;

exports[`Transpile: NodeJS v8 react: jsx 1`] = `
"\\"use strict\\";

function render() {
  return React.createElement(\\"h1\\", null, \\"Hello\\");
}"
`;

exports[`Transpile: NodeJS v8 react: jsx spread 1`] = `
"\\"use strict\\";

function render(props) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, props), \\"Hello\\");
}"
`;

exports[`Transpile: NodeJS v8 react: proptypes 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _react = _interopRequireDefault(require(\\"react\\")),
    _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

class MyComponent extends _react.default.Component {
  render() {
    return _react.default.createElement(\\"div\\", null, this.props.userName);
  }

}

MyComponent.propTypes = {
  userName: _propTypes.default.string
};"
`;

exports[`Transpile: NodeJS v8 require context 1`] = `
"\\"use strict\\";

if (process.env.NODE_ENV === \\"test\\") {
  require('babel-plugin-require-context-hook/register')();
}

const loader = __requireContext(__dirname, __dirname, false, /\\\\.js/);"
`;
